src/metabase/email/messages.clj

```diff
@@ -310,7 +321,14 @@
            ;; There are legacy pulses that exist without being tied to a dashboard
            :dashboardHasTabs          (when dashboard-id (dashboard/has-tabs? dashboard-id))
            :creator                   (-> pulse :creator :common_name)
-           :sectionStyle              (style/style (style/section-style))}
+           :sectionStyle              (style/style (style/section-style))
+           :notificationText          (if (nil? non-user-email)
+                                        "Manage your subscriptions"
+                                        "Unsubscribe")
+           :notificationManagementUrl (if (nil? non-user-email)
+                                        (urls/notification-management-url)
+                                       ;; TODO: change this to whatever FE URL we chooseg
+                                        (str (urls/notification-management-url) "?hashdata=" (generate-pulse-unsubscribe-hash (:id pulse) non-user-email)))}
```

Comment:
I've written `generate-pulse-unsubscribe-hash` to include all of them, but I could split them into different params. Thoughts on which way is better? I guess splitting them is cleaner

Reply:
Hash validation happens by taking the pulse id and email and generating a new hash and checking if it's the same, right? So the FE needs access to the id and email separately from the hash so that it can include them in the API call, right?

Or am I misunderstanding something?

Reply:
The function `generate-pulse-unsubscribe-hash` returns a JSON string which includes hash, pulse id, and email